ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,165 @@
+    {
+        //GameObject[] arrayofspheres = GameObject.FindGameObjectsWithTag("Sphere");
+        table = GameObject.Find("Table");
+        tableBounds = table.GetComponent<Collider>().bounds;
+        targetY = transform.position.y;
+    }
+
+
+    // CONTROL
+
+    public void Move(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        // random position (on table, within reach)     
+        Vector2 tableTopPoint = RandomReachablePointOnTable(listOfAlreadyMovedObjects);
+        Vector3 tableCenter = tableBounds.center;
+        float x = tableCenter.x + tableTopPoint.x;
+        float z = tableCenter.z + tableTopPoint.y;
+        transform.position = new Vector3(x, targetY, z);
+
+        // random rotation
+        Vector3 randomRotation = new Vector3(
+            transform.rotation.eulerAngles.x,
+            Random.value * 360.0f,
+            transform.rotation.eulerAngles.z);
+        transform.rotation = Quaternion.Euler(randomRotation);
+    }
+
+
+    // HELPERS
+
+    Vector2 RandomReachablePointOnTable(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        while (true)
+        {
+            Vector2 randomOffset = RandomPoint(robotMinReach, robotMaxReach, listOfAlreadyMovedObjects);
+            bool onTable = PointOnTable(randomOffset);
+            if (onTable)
+            {
+                return randomOffset;
+            }
+        }
+    }
+
+    bool PointOnTable(Vector2 point)
+    {
+        /*  point: The 2D point on table top, relative to center of table top.
+         *  Determines if this point would be on the table or not.      
+         */
+        Vector3 tableExtents = tableBounds.extents;
+        float targetRadius = 0f;
+        if (tag == "Sphere"){
+                targetRadius = GetComponent<SphereCollider>().radius * GetComponent<Transform>().localScale[0]; // for the sphere 
+            }
+            // otherwise it is a cylinder 
+            else{
+                targetRadius = GetComponent<CapsuleCollider>().radius * GetComponent<Transform>().localScale[0]; 
+            }
+
+        float safeXDistance = tableExtents.x - targetRadius;
+        float safeZDistance = tableExtents.z - targetRadius;
+        float xDistance = Mathf.Abs(point.x);
+        float yDistance = Mathf.Abs(point.y);
+        bool onTable = (xDistance < safeXDistance) && (yDistance < safeZDistance);
+        return onTable;
+    }
+
+    public Vector2 RandomPoint(float minRadius, float maxRadius, List<GameObject> listOfAlreadyMovedObjects)
+    {
+        /*  Picks a 2D point randomly at uniform. Must be between minRadius and
+         *  maxRadius from center and not at the same place than the cube. 
+         *  Point given relative to center.      
+         */
+        //for (var j = 0; j < 100; j++)
+        while (true)
+        {
+            // pick a random point between the square of edge 2*minRadius and the one with edge 2*maxRadius to be sure that the point is reachable by the robot 
```

Comment:
This will not guarantee that the point is reachable - the corners of the square are beyond the maxRadius.